lib/Module.php

```diff
@@ -50,17 +85,22 @@ public function __construct(string $filePath)
             throw new RuntimeException("File `$filePath` is not readable.");
         }
 
-        $wasmBytes = wasm_read_bytes($filePath);
-
-        if (null === $wasmBytes) {
-            throw new RuntimeException("An error happened while reading the module `$filePath`.");
+        $wasmBytes = wasm_fetch_bytes($filePath);
+        $wasmModuleUniqueIdentifier = null;
+
+        if (self::PERSISTENT === $persistence) {
+            $wasmModuleUniqueIdentifier = $this->getUniqueIdentifier($filePath);
+        } else {
+            // Validating WebAssembly bytes forces the `wasm_bytes` resource to
+            // be read. In the case of a persistent module, we don't want to
+            // read the bytes. That's why `wasm_validate` is called only when
+            // the module is not persistent.
+            if (false === wasm_validate($wasmBytes)) {
```

Comment:
I think the compile functions also validate so this might be validating twice if compile is called below.

Reply:
You read in my mind. It was my question when I wrote this comment! Thanks to confirm!